common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -633,8 +649,57 @@ def get_problem_html(self, encapsulate=True, save_notification_message=None):
         # If demand hints are available, emit hint button and div.
         demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
         demand_hint_possible = len(demand_hints) > 0
+        # hint_index is the index of the last hint that will be displayed in this rendering,
+        # so add 1 to check if others exist.
+        should_enable_next_hint = demand_hint_possible and hint_index + 1 < len(demand_hints)
+
+        answer_notification_type, answer_notification_message = self._get_answer_notification()
```

Comment:
Note to reviewers: this method became too long and produced a pylint/pep8 warning. I pulled the answer notification code into a helper method, _get_answer_notification.